proposals/0000-amending-mrp.rst

```diff
@@ -0,0 +1,256 @@
+      Remove definition for ‘return’
+      See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
+    |
+  7 |     return = MkId
+    |     ^^^^^^^^^^^^^
+
+  example.hs:8:5: error: [-Wnoncanonical-monad-instances]
+      Noncanonical ‘(>>)’ definition detected
+      in the instance declaration for ‘Monad Id’.
+      ‘(>>)’ will eventually be removed in favour of ‘(*>)’
+      Remove definition for ‘(>>)’, and implement ‘(*>)’ with an efficient definition.
+      See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
+    |
+  8 |     (>>) _ b = b
+    |     ^^^^^^^^^^^^
+
+  example.hs:12:5: error: [-Wredundant-canonical-monad-instances]
+      ‘return’ definition detected
+      in the instance declaration for ‘Monad Id’.
+      ‘return’ will eventually be removed in favour of ‘pure’
+      Remove definition for ‘return’
+      See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
+    |
+  12|     return = pure
+    |     ^^^^^^^^^^^^^
+
+  example.hs:13:5: error: [-Wredundant_canonical-monad-instances]
+      ‘(>>)’ definition detected
+      in the instance declaration for ‘Monad Id’.
+      ‘(>>)’ will eventually be removed in favour of ‘(*>)’
+      Remove definition for ‘(>>)’, and implement ‘(*>)’ with an efficient definition.
+      See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return
+    |
+  13|     (>>) = (*>)
+    |     ^^^^^^^^^^^
+
+Effect and Interactions
+-----------------------
+Speeding up the phases of the proposal means that we will eventually be rid of
+the warnings and errors we are building up in service to this proposal.
+Encouraging users to implement ``(*>)`` efficiently will also mean that more
+programs are likely to be performant.
+
+The alternative is that we warn against a change that we are not intending on 
+making.
+
+
+Costs and Drawbacks
+-------------------
+rhendric has already made a MR to perform phase 2 of the amended proposal, which
+can be found `here <https://gitlab.haskell.org/ghc/ghc/-/merge_requests/13999>`_.
+As such the only code cost would be for stage 3 of the amended proposal.
+
+Old tutorials may become more incorrect as they will no longer be able to define
+``Monad`` fully.
+
+Backward Compatibility
+----------------------
+This amendment would mean that we have breakage in uncommon cases. In one 
+attempt to compile Stackage with a patched GHC, ~94% of packages that were 
+attempted compiled. However, there are a handful of packages that do not want
+to accept fixes that would make them compatible with the changes in this 
+proposal which are depended upon by hundereds of other packages, which makes
+further analysis hard.
+
+However, the compiler error that is coming as a result of this proposal has been
+a warning since 2015, and has been on by default since 9.2.1 in 2020, so there
+shouldn't need to be more lead in.
```

Comment:
If we do as Roman proposes and change the default implementation of `(*>)` to have a `Monad` constraint, that would likely break more packages. (We'd need to have a new preliminary warning schedule for that change, I think, inserting a new phase to the plan.)